main.go

```diff
@@ -152,7 +153,19 @@ func main() {
 
 	// adds the upgrade mechanism to be executed once the manager is ready
 	err = mgr.Add(manager.RunnableFunc(func(c context.Context) error {
-		return upgrade.ManagedInstances(c, ctrl.Log.WithName("upgrade"), v, mgr.GetClient())
+		return collectorupgrade.ManagedInstances(c, ctrl.Log.WithName("collector-upgrade"), v, mgr.GetClient())
```

Comment:
This might be a good opportunity to refactor this, placing the upgrade routines in a bootstrap/setup procedure somewhere. The main.go would then just call the bootstrap's `RunnableFunc`.

Reply:
happy to do it in a separate refactoring PR